src/plugins/csp_configuration_provider/server/provider.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { OpenSearchClient } from '../../../../src/core/server';
+
+import { CspClient } from './types';
+
+export class OpenSearchCspClient implements CspClient {
+  private client: OpenSearchClient;
+
+  constructor(inputOpenSearchClient: OpenSearchClient) {
+    this.client = inputOpenSearchClient;
+  }
+
+  async exists(configurationName: string): Promise<boolean> {
+    let value = false;
+
+    try {
+      const exists = await this.client.indices.exists({
+        index: configurationName,
+      });
+
+      value = exists.body;
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error(
+        `Failed to call exists with configurationName ${configurationName} due to error ${e}`
+      );
+    }
+
+    return value;
+  }
+
+  async get(configurationName: string, cspRulesName: string): Promise<string> {
+    const query = {
+      query: {
+        match: {
+          _id: {
+            query: cspRulesName,
+          },
+        },
+      },
+    };
+
+    let value = '';
+
+    try {
+      const data = await this.client.search({
```

Comment:
Is doc id uniqued, if so why not use doc api instead of search api

Reply:
will switch to doc api